### DISCUSSION TITLE
BUFR in Met-9.1.3

### DISCUSSION THREAD

**Discussion by AnastasiaBundel** [dtcenter/METplus#995] [File I/O] [answered]
Hello! 
At https://dtcenter.org/book/export/html/1155  it is written that "The PB2NC tool may be run on both PrepBufr and Bufr observation files. As of met-6.1, support for Bufr is limited to files containing embedded tables. Support for Bufr files using external tables will be added in a future release."
**Is BUFR format with external tables supported in MET version 9.1.3?**
We tried to run PB2NC with a BUFR file created using the ECMWF tables. 
But we got an error:

*** Running PB2NC on a PREPBUFR file ***
DEBUG 1: Default Config File: /cosmo/akirsanov/MET/share/met/config/PB2NCConfig_default
DEBUG 1: User Config File: /cosmo/abundel/MET/scripts/PB2NC/PB2NCConfig_G212
DEBUG 1: Pre-processing BUFR File for variable names from /cosmo/abundel/BUFR/synop2021062300.bufr
 +++++++++++++++++++++WARNING+++++++++++++++++++++++
 BUFRLIB: OPENBT - THIS IS A DUMMY BUFRLIB ROUTINE CALLED BY CKTABA OR APPL. PGM; OPENBT SHOULD BE INCL. IN-LINE IN APPL. PGM
 +++++++++++++++++++++WARNING+++++++++++++++++++++++

 +++++++++++++++++++++WARNING+++++++++++++++++++++++
 **BUFRLIB: CKTABA - UNRECOGNIZED TABLE A MESSAGE TYPE (FN000000) - RETURN WITH IRET = -1**
 +++++++++++++++++++++WARNING+++++++++++++++++++++++

DEBUG 1: process_pbfile_metadata() -> **the number of records: 0**
WARNING:
WARNING: process_pbfile_metadata() -> No Bufr messages to process in file: /cosmo/abundel/BUFR/synop2021062300.bufr
WARNING:
DEBUG 1:
DEBUG 1:    Header variables (0) :
DEBUG 1:
DEBUG 1:    Observation variables (0)  Name: Description                       Types:

**Comment by PerryShafran-NOAA** [comment]
This is a good question.  I'm not seeing anything in any of the documentation that suggests that you can run METplus with an external prepbufr table.  All our runs have been with bufr files that contain the tables within them.  I don't even see attaching an external table as an option.  

Can you describe the pb2nc command used to run pb2nc?  How did you identify the external table?  

Perry

**Reply by AnastasiaBundel** [reply]
Hi @PerryShafran-NOAA, thank you for the feedback!
We actually don't identify the external tables. There are descriptor bytes in the file, and in the external table, the correspondense is set up with the variable corresponding to the descriptor, and other relevant metadata, as far as I understand. In the VERSUS verification package we are using at present with bufrs, the ECMWF tables are referenced from within.

The command to run the PB2NC is pretty standard, taken from the example. We've just begun to explore MET,
Here it is (if this is what you meant): 
pb2nc \
~/BUFR/synop2021062300.bufr \
 ${TEST_OUT_DIR}/PB2NC/sample_pb.nc \
~/MET/scripts/PB2NC/PB2NCConfig_G212 \
-v 2 -index

**Comment by jprestop** [comment, marked as answer]
Hi @AnastasiaBundel.  Thank you for your question and for your interest in MET.  @PerryShafran-NOAA, thank you for your help here as well.  Currently, MET does not have support for ECMWF BUFR data using external tables.  However, we do have an existing GItHub Issue, [Add support for ECMWF BUFR data using external tables. #926](https://github.com/dtcenter/MET/issues/926), to add this functionality.  We are hoping to have this functionality added in MET-10.1.0, but I am unsure if we are on track for that release.  I will follow up with the developer and will reply here as soon as I have more information.

**Reply by AnastasiaBundel** [reply]
Hi @jprestop ! Thank you for the help and the link to the issue! Got it, it would be great if this functionality were added in MET.
Meanwhile,  the alternative is to transfer our point observations to ASCII and use ASCII2NC, I guess?

**Reply by jprestop** [reply]
@AnastasiaBundel, if you are able to transfer your point observations to ASCII and use ASCII2NC, I do think that would be good temporary solution.

**Reply by AnastasiaBundel** [reply]
We will try. Thanks!

**Reply by jprestop** [reply]
@AnastasiaBundel I just wanted to follow up and let you know our goal is to have this functionality added for MET-10.1.0.

**Reply by AnastasiaBundel** [reply]
Thank you, @jprestop! When is it planned to release it, approximately, couldn't you tell?

**Reply by jprestop** [reply]
You bet!  My apologies for not stating that information.  It should be in the December 2021/January 2022 timeframe.

**Reply by TaraJensen** [reply]
But it will be available in a beta release in Sept/Oct time frame.  Maybe @AnastasiaBundel you can help us out by testing the capability once it's added?

**Reply by AnastasiaBundel** [reply]
Got it, thank you!

**Reply by AnastasiaBundel** [reply]
@TaraJensen, Yes, voluntary!
